File: Minyoung06/PGS_12909.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        Deque<String> stack = new ArrayDeque<>();
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='(')
                stack.push("(");
            else{
                if(stack.isEmpty()) return false;
                stack.pop();
            }
        }
        if(stack.isEmpty())return true;
        else return false;
    }
}

File: Minyoung06/PGS_12909.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    int balance=0;
    for(char c:s){
        if(c=='('){
            balance++;
        }
        else if(c==')'){
            balance--;
            if(balance<0) return false;
        }
        else{return false;}
    }
    if(balance!=0) return false;
    return true;
}